### PR TITLE
Make FunctionPresetBC incremental if required

### LINKED ISSUE #11559: Apply a FunctionPresetBC or FunctionDirichletBC incrementally
## Rationale

In several problems for tensor mechanics we need to increase the displacements on a boundary of the specimen starting from a preloaded initial configuration (either through other BCs or through eigenstrains), which is generally unknown. However, the `FunctionPresetBC` and `FunctionDirichletBC` impose the new value of the displacement without considering the previous state of the material. We need to be able to evaluate the BC incrementally such as: 

`u(t_n+1) = u(t_n) + g(t_n+1) - g(t_n)`

Which might be different from `g(t_n+1)` if the BC was not active at `t_n` or before.

## Description

Add an `incremental` boolean flag (default false) that would let the BC to be computed incrementally if requested by the user. If false, we fall back on the existing version `u(t_n+1) = g(t_n+1)`

## Impact

More versatile `FunctionPresetBC` and `FunctionDirichletBC`.

### PR DESCRIPTION
**Pull request by agiorla**
Both FunctionPresetBC and FunctionDirichletBC can be computed incrementally if required by the user.

Closes #11559

### CODE REVIEW

**Comment by agiorla** [comment]
Tagging @bwspenc

File: framework/src/bcs/FunctionPresetBC.C [hunk 1]
```diff
@@ -18,18 +18,29 @@ validParams<FunctionPresetBC>()
...
+    _func(getFunction("function")),
+    _incremental(getParam<bool>("incremental")),
+    _u_old(_incremental ? &_var.dofValuesOld() : nullptr)
 {
 }
 
 Real
 FunctionPresetBC::computeQpValue()
 {
+  if (_incremental)
```
[the remark this answers is not in the document]
Comment (permcody) [reply framework/src/bcs/FunctionPresetBC.C]: I like the idea of a new rate based BC. That might even be general enough for the framework.

[the remark this answers is not in the document]
Comment (agiorla) [reply framework/src/bcs/FunctionPresetBC.C]: OK, I'll make it as separate classes no problem. Should I close this PR and start a new one or continue pushing to that branch?

[the remark this answers is not in the document]
Comment (permcody) [reply framework/src/bcs/FunctionPresetBC.C]: Either is fine!

[the remark this answers is not in the document]
Comment (agiorla) [reply framework/src/bcs/FunctionPresetBC.C]: I find it easier to just start a new one

**Pull request closed without merging**